sema/commons/j2/j2_functions.py

```diff
@@ -81,6 +81,15 @@ def xsd_format_date(content, quote, suffix):
         asdt = content
     return xsd_value(asdt.isoformat(), quote, "xsd:date")
 
+def xsd_format_gyear(content, quote, suffix):
+    # make rigid gYear
+    assert (isinstance(content, int) and 1000 <= content <= 9999), "year should be 4-digit integer"
+    return xsd_value(content, quote, "xsd:gYear")
+
+def xsd_format_gmonthyear(content, quote, suffix):
+    # make rigid gMonthYear
```

Comment:
same remark on multiple inputs:

in this case it would make sense to support input in date and datetime format and forcefully convert it to YYYY-MM format